Jenkinsfile

```diff
@@ -0,0 +1,137 @@
+            def chartManager = new io.stakater.charts.ChartManager()
+            def docker = new io.stakater.containers.Docker()
+            def stakaterCommands = new io.stakater.StakaterCommands()
+            def slack = new io.stakater.notifications.Slack()
+            try {
+                stage('Download Dependencies') {
+                    sh """
+                        cd ${srcDir}
+                        make install
+                        cp -r ./vendor/* /go/src/
+                    """
+                }
+
+                stage('Run Tests') {
+                    sh """
+                        cd ${srcDir}
+                        make test
+                    """
+                }
+
+                if (utils.isCI()) {
+                    stage('CI: Publish Dev Image') {
+                        dockerImageVersion = stakaterCommands.getBranchedVersion("${env.BUILD_NUMBER}")
+                        
+                        sh """
+                            cd ${srcDir}
+                            export DOCKER_IMAGE=${dockerImage}
+                            export DOCKER_TAG=${dockerImageVersion}
+                            make binary-image
+                            make push
+                        """
+                    }
+                    
+                    stage('Notify') {
+                        def dockerImageWithTag = "${dockerImage}:${dockerImageVersion}"
+                        slack.sendDefaultSuccessNotification(slackWebHookURL, slackChannel, [slack.createDockerImageField(dockerImageWithTag)])
+
+                        def commentMessage = "Image is available for testing. `docker pull ${dockerImageWithTag}`"
+                        git.addCommentToPullRequest(commentMessage)
+                    }
+                } else if (utils.isCD()) {
+                    stage('CD: Tag and Push') {
+                        print "Generating New Version"
+                        def version = common.shOutput("jx-release-version --gh-owner=${repoOwner} --gh-repository=${repoName}")
+                        dockerImageVersion = version
+                        //TODO: jx-release-version won't read bumped version from .version
+                        sh """
+                            echo "${version}" > .version
+                        """
+
+                        // Render chart from templates
+                        templates.renderChart(chartTemplatesDir, chartDir, repoName, version, dockerImage)
+                        // Generate manifests from chart
+                        templates.generateManifests(chartDir, repoName, manifestsDir)
+
+                        git.commitChanges(WORKSPACE, "Bump Version to ${version}")
+
+                        print "Pushing Tag ${version} to Git"
+                        git.createTagAndPush(WORKSPACE, version)
+                        git.createRelease(version)
+
+                        print "Pushing Tag ${version} to DockerHub"
```

Comment:
@waseem-h shouldn't this be done before we tag the git repo? it seems quite a bit misplaced